### DISCUSSION TITLE
Run as daemon w/o systemd

### DISCUSSION THREAD

**Discussion by reesericci** [coder/code-server#2216] [Q&A] [answered]
I would like to be able to launch code-server with a command like `code-server -d` to launch it as a daemon without using systemctl services. (Like in a docker container, or systems without systemd)

**Comment by nhooyr** [comment, marked as answer]
You can do `code-server & && disown` to run it in the background from a shell and keep it running after the shell exits.

I'd recommend using `tmux` or the like anyway so that you can easily restart and record output.

**Comment by CooperSma** [comment]
+1

**Comment by reesericci** [comment]
will anyone work on this?

**Comment by code-asher** [comment]
We currently don't have plans to implement this (although we might in
the future since this is an idea we've thought about in the past) but
PRs are welcome!

**Comment by reesericci** [comment]
yea, i don't know typescript :(

**Comment by nhooyr** [comment]
@reesericci could you clarify why `code-server & && disown` isn't enough? Or using `tmux` or `screen` so that you have easy access to the logs/output.

**Comment by reesericci** [comment]
I'm using it in a docker container with other services, I don't use docker-compose.

**Comment by nhooyr** [comment]
If you start it with a shell, you can do `code-server & && disown` to put it in the background.
